### PR TITLE
[FLINK-30485][Connector/RabbitMQ] Remove RabbitMQ connector

### PR DESCRIPTION
**Pull request by MartijnVisser**
## What is the purpose of the change
The RabbitMQ connector has been externalized to https://github.com/apache/flink-connector-rabbitmq/

### CODE REVIEW

File: .idea/vcs.xml [hunk 1]
```diff
@@ -1,5 +1,11 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
+  <component name="CommitMessageInspectionProfile">
+    <profile version="1.0">
+      <inspection_tool class="CommitFormat" enabled="true" level="WARNING" enabled_by_default="true" />
+      <inspection_tool class="CommitNamingConvention" enabled="true" level="WARNING" enabled_by_default="true" />
+    </profile>
+  </component>
```
Comment (XComp) [inline .idea/vcs.xml:8]: Added accidentally?

**Review by XComp — COMMENTED** [review]
I'll wait with further reviewing this PR until we sorted out PR #21536. It appears that I have similar questions here. :-D

File: .idea/vcs.xml (hunk 1 above)
Comment (MartijnVisser) [reply .idea/vcs.xml:8]: Mehhh yes, the new IntelliJ keeps changed it

File: flink-python/pom.xml [hunk 2]
```diff
@@ -298,7 +298,7 @@ under the License.
 			<!-- Indirectly accessed in pyflink_gateway_server -->
 			<groupId>org.apache.flink</groupId>
 			<artifactId>flink-sql-connector-rabbitmq</artifactId>
-			<version>${project.version}</version>
+			<version>3.0.0-1.16</version>
```
Comment (XComp) [inline flink-python/pom.xml:301]: That's related to [my comment in the jdbc removal PR](https://github.com/apache/flink/pull/21536#discussion_r1056276909)

File: docs/data/sql_connectors.yml [hunk 3]
```diff
@@ -196,4 +196,4 @@ rabbitmq:
     name: RabbitMQ
     category: connector
     maven: flink-connector-rabbitmq
-    sql_url: https://repo.maven.apache.org/maven2/org/apache/flink/flink-sql-connector-rabbitmq/$version/flink-sql-connector-rabbitmq-$version.jar
+    sql_url: https://repo.maven.apache.org/maven2/org/apache/flink/flink-sql-connector-rabbitmq/3.0.0-1.16/flink-sql-connector-rabbitmq-3.0.0-1.16.jar
```
Comment (XComp) [inline docs/data/sql_connectors.yml:199]: :+1:

**Review by XComp — APPROVED** [review]
LGMT :+1: Only this one thing about updating the dependency version for PyFlink is open

**Comment by MartijnVisser** [comment]
@flinkbot run azure

**Pull request merged**